src/Rings/mpoly-ideals.jl

```diff
@@ -180,7 +180,10 @@ function Base.:*(I::MPolyIdeal{T}, J::MPolyIdeal{T}) where T
       end
     end
   end
-  return ideal(base_ring(I), unique!(newgens))
+  result = ideal(base_ring(I), unique!(newgens))
+  # carry over eventual information about the dimension
+  is_known(I, dim) && is_known(J, dim) && (result.dim = maximum([dim(I), dim(J)]))
```

Comment:
Matter of taste, but I personally think assignments in a chain of `&&` takes it a step to far in terms of how it affects code clarity. But feel free to ignore this suggestion *shrug*
```suggestion
  if is_known(I, dim) && is_known(J, dim)
    result.dim = maximum([dim(I), dim(J)]))
  end
```